server/py/framework/db/sqldb/db.py

```diff
@@ -7074,6 +7075,23 @@ def delete_model_endpoints(
             main_table_identifier_values=uids,
         )
 
+    def get_system_id(self, session: Session) -> typing.Optional[str]:
+        system_id_record = (
+            self._query(session, SystemMetadata)
+            .filter(SystemMetadata.key == "system_id")
+            .one_or_none()
+        )
+        if not system_id_record:
+            logger.debug("System id not found in DB")
+            return None
+        return system_id_record.value
```

Comment:
```suggestion
        return system_id_record.value if system_id_record else None
```

not sure the log needs to happen here or by its callee